File: backend/internal/model/db/projection/prepaid_category.go
package projection

//dbmap:sqlcdb=PrepaidCategory
type PrepaidCategory struct {
	ID               int64   `db:"id" json:"id"`
	CategoryUUID     string  `db:"category_uuid" json:"category_uuid"`
	MerchantID       int64   `db:"merchant_id" json:"merchant_id"`
	Name             string  `db:"name" json:"name"`
	AccountID        string  `db:"account_id" json:"account_id"`
	ExpenseAccountID string  `db:"expense_account_id" json:"expense_account_id"`
	IsActive         bool    `db:"is_active" json:"is_active"`
	UpdatedBy        *string `db:"updated_by" json:"updated_by"`
	UpdatedAt        string  `db:"updated_at" json:"updated_at"`
	Version          int64   `db:"version" json:"version"`
}

File: frontside/src/types/prepaidCategory.ts
export interface PrepaidCategory {
  id:               number;
  category_uuid:    string;
  merchant_id:      number;
  name:             string;
  account_id:       string;
  expense_account_id: string;
  is_active:        boolean;
  updated_by:       string | null;
  updated_at:       string;
  version:          number;
}

export interface PrepaidCategoryCreatePayload {
  name:               string;
  account_id:         string;
  expense_account_id: string;
}

export interface PrepaidCategoryUpdatePayload {
  expected_version:   number;
  name:               string;
  account_id:         string;
  expense_account_id: string;
}
